### PR TITLE
Finetuning

### PR DESCRIPTION
**Pull request by Godofnothing**
Added `--finetune_keep_best` option and fixed loading in `modelutils.py`

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -649,40 +645,51 @@ def update_outs_parallel(
         "--finetune_max_epochs",
         type=int,
         default=1000,
-        help="Run this many passes over training data when doing global optimization; no means skip GO",
+        help="Run this many passes over training data when doing finetuning; no means skip",
```
Comment (BlackSamorez) [inline main.py:648]: "0 means skip"?

File: src/finetune.py [hunk 2]
```diff
@@ -122,6 +129,11 @@ def finetune_groupwise(
 
         if args.finetune_relative_mse_tolerance is not None:
             epoch_loss = loss_numerator / loss_denominator
+            if args.finetune_keep_best:
+                if epoch_loss / previous_best_loss < 1.0:
+                    best_parameters = deepcopy(differentiable_parameters)
+                else:
+                    differentiable_parameters = best_parameters
```
Comment (BlackSamorez) [inline src/finetune.py:136]: What's the point of this? If loss increases we roll back to best?

File: main.py (hunk 1 above)
Comment (Godofnothing) [reply main.py:648]: This means, that we do not finetune. I can write "no means skip finetune"

File: src/finetune.py (hunk 2 above)
Comment (Godofnothing) [reply src/finetune.py:136]: Yes, seems to bring marginal improvement.

File: main.py (hunk 1 above)
Comment (BlackSamorez) [reply main.py:648]: It's an `int` argument. It can't be 'no'

Comment (Godofnothing) [reply main.py:648]: Agree, I'll fix it

**Review by Vahe1994 — APPROVED** [review]
